packages/flutter/lib/src/painting/shape_decoration.dart

```diff
@@ -237,12 +237,31 @@ class ShapeDecoration extends Decoration {
       if (t == 1.0)
         return b;
     }
+
+    Color? lerpColor = Color.lerp(a?.color, b?.color, t);
+    Gradient? lerpGradient = Gradient.lerp(a?.gradient, b?.gradient, t);
+    final DecorationImage? lerpImage = t < 0.5 ? a!.image : b!.image; // TODO(ianh): cross-fade the image
+    final List<BoxShadow>? lerpShadows = BoxShadow.lerpList(a?.shadows, b?.shadows, t);
+    final ShapeBorder lerpShape = ShapeBorder.lerp(a?.shape, b?.shape, t)!;
+
+    // The [color] and [gradient] properties are mutually exclusive, one (or
+    // both) of them must be null.
+    if (lerpColor != null && lerpGradient != null) {
+      if (t < 0.5) {
```

Comment:
It is unclear to me whether this is the correct way of lerping, there will be a sudden jump at t ==0.5

Is it possible create a new gradient where all its colors lerp with the other color. Otherwise, we should probably just throw and say it is not supported

Reply:
Image and boolean makes sense to have this jump. Let's ask in the issue first before we proceed